pkg/controller/nodes/task/catalog/datacatalog/transformer.go

```diff
@@ -131,16 +134,16 @@ func GenerateArtifactTagName(ctx context.Context, inputs *core.LiteralMap) (stri
 // Get the DataSetID for a task.
 // NOTE: the version of the task is a combination of both the discoverable_version and the task signature.
 // This is because the interfact may of changed even if the discoverable_version hadn't.
-func GenerateDatasetIDForTask(ctx context.Context, task *core.TaskTemplate) (*datacatalog.DatasetID, error) {
-	datasetVersion, err := generateDataSetVersionFromTask(ctx, task)
+func GenerateDatasetIDForTask(ctx context.Context, k catalog.Key) (*datacatalog.DatasetID, error) {
+	datasetVersion, err := generateDataSetVersionFromTask(ctx, k.Type, k.TypedInterface, k.CacheVersion)
```

Comment:
@EngHabu @kumare3 when running this on staging, the `catalog.Key.CacheVersion` is empty.

https://kibana-staging.lyft.net/goto/aa22b6c62a1ff11549c691535110d07c